GauchoGrubAndroid/app/src/main/java/com/g10/gauchogrub/MenuFragment.java

```diff
@@ -5,12 +5,136 @@
+import android.os.AsyncTask;
+import java.io.File;
+import java.util.ArrayList;
+
+import com.g10.gauchogrub.menu.DayMenu;
+
+
+public class MenuFragment extends Fragment implements AdapterView.OnItemSelectedListener, Runnable {
+
+
+    private TableLayout menuTable;
+
+    public final static Logger logger = Logger.getLogger("MenuFragment");
+
+    private String diningCommon;
+    private String date;
 
-public class MenuFragment extends Fragment {
 
     @Override
     public View onCreateView(LayoutInflater inflater, ViewGroup container, Bundle savedInstanceState) {
         View rootView = inflater.inflate(R.layout.menu_fragment, container, false);
+
+        this.menuTable = (TableLayout) rootView.findViewById(R.id.menu_table);
+
+        // Initialize Spinner (DC)
+        Spinner spinner = (Spinner)rootView.findViewById(R.id.dining_common_spinner);
+        ArrayAdapter<CharSequence> adapter = ArrayAdapter.createFromResource(getActivity(), R.array.dining_commons_array, R.layout.dining_cams_spinner_item);
+        adapter.setDropDownViewResource(R.layout.dining_cams_spinner_item);
+        spinner.setAdapter(adapter);
+        spinner.setOnItemSelectedListener(this);
+        spinner.setSelection(0);
+        // Initialize Spinner (date)
+        Spinner dateSpinner = (Spinner)rootView.findViewById(R.id.date_spinner);
+        ArrayAdapter<CharSequence> dateAdapter = ArrayAdapter.createFromResource(getActivity(), R.array.date_nav_array, R.layout.dining_cams_spinner_item);
+        dateAdapter.setDropDownViewResource(R.layout.dining_cams_spinner_item);
+        dateSpinner.setAdapter(dateAdapter);
+        dateSpinner.setOnItemSelectedListener(this);
+        dateSpinner.setSelection(0);
+
         return rootView;
     }
+
+    private void inflateMenu(DayMenu displayMenu) {
+
+        menuTable.removeAllViews();
+
+        ArrayList<Meal> menuMeals = displayMenu.getMeals();
+
+        if(displayMenu.getMeals() == null) return;
+        for (Meal headerEntry : menuMeals) {
+            String mealName = headerEntry.getMealName();
+
+            TableRow headerRow = new TableRow(getActivity().getApplicationContext());
+            headerRow.setLayoutParams(new TableRow.LayoutParams(TableRow.LayoutParams.MATCH_PARENT, TableRow.LayoutParams.WRAP_CONTENT));
+            View headerView = LayoutInflater.from(getActivity().getApplicationContext()).inflate(R.layout.meal_header, null);
+            TextView headerTextView = (TextView) headerView.findViewById(R.id.header);
+            headerTextView.setText(mealName);
+            headerRow.addView(headerView);
+            menuTable.addView(headerRow);
+
+            for(MenuItem itemEntry : headerEntry.getMenuItems()) {
+                String itemTitle = "   " + itemEntry.getTitle();
+                String itemCategory = itemEntry.getMenuCategory();
+                String itemType = itemEntry.getMenuItemType();
+
+                if(itemType.equals("Regular")) itemType = "";
```

Comment:
Please, always use curly brackets. Code is hard to read otherwise.
For this particular line you can say itemType = (itemType.equals("Regular")) ? "" : itemType